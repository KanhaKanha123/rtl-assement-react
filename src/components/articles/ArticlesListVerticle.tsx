import ArticleHorizontalLayout from "./layout/ArticleHorizontalLayout";
import { BundlesProps } from "../../types/types";

type ArticlesListVerticleProps = {
    bundles: BundlesProps[];
};

const ArticlesListVerticle = ({ bundles }: ArticlesListVerticleProps) => {
    return  (<div data-testid="verticleArticles-wrapper" className="verticle-articles">
    {(bundles || []).map((article: any) => (<ArticleHorizontalLayout key={article.id} bundle={article}></ArticleHorizontalLayout>))}
    </div>)
};

export default ArticlesListVerticle;