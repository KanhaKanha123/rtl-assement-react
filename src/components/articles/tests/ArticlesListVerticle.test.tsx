import ArticlesListVerticle from '../ArticlesListVerticle';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import * as testData from './test.data.json';

describe('ArticlesListVerticle component test', () => {
    const { bundles } = testData;
    beforeEach(() => {
        render(<ArticlesListVerticle bundles={bundles} />);
    });

    test("ArticlesListVerticle component load correctly", () => {
        const wrapper = screen.getByTestId("verticleArticles-wrapper");

        //check element exist
        expect(wrapper).toBeInTheDocument();
    });

    test("ArticlesListVerticle component should match snapshot", () => {
        const element = renderer.create(<ArticlesListVerticle bundles={bundles} />);
        expect(element.toJSON()).toMatchSnapshot();
    });
});
