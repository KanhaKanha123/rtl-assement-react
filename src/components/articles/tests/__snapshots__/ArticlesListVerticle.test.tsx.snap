// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesListVerticle component test ArticlesListVerticle component should match snapshot 1`] = `
<div
  className="verticle-articles"
  data-testid="verticleArticles-wrapper"
>
  <a
    data-testid="horizontalLayout-a"
    href="https://www.rtlboulevard.nl/entertainment/showbizz/artikel/4245471/trijntje-oosterhuis-onherkenbaar-veel-afgevallen"
    target="_blank"
  >
    <div
      className="article-horizontal-layout"
      data-testid="horizontalLayout-wrapper"
    >
      <img
        alt="De uitschieters van 2018"
        className="card-img-top"
        data-testid="horizontalLayout-img"
        src="https://www.rtlboulevard.nl/sites/default/files/content/images/2018/06/19/ANP-56919532.jpg"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="right-content-div"
        data-testid="horizontalLayout-rightContentDiv"
      >
        <div
          className="label card-title h5"
          data-testid="horizontalLayout-label-title"
        >
          De uitschieters van 2018
        </div>
        <div
          className="label card-title h5"
          data-testid="horizontalLayout-label-date"
        >
          19 juni 2018
        </div>
        <span
          className="layout-bold-title"
          data-testid="horizontalLayout-layoutBoldTitle"
        >
          Trijntje
        </span>
        <span
          className="lead"
          data-testid="horizontalLayout-lead"
        >
          Humberto Tan
        </span>
      </div>
    </div>
  </a>
  <a
    data-testid="horizontalLayout-a"
    href="https://www.rtlboulevard.nl/entertainment/showbizz/artikel/4245471/trijntje-oosterhuis-onherkenbaar-veel-afgevallen"
    target="_blank"
  >
    <div
      className="article-horizontal-layout"
      data-testid="horizontalLayout-wrapper"
    >
      <img
        alt="De uitschieters van 2018"
        className="card-img-top"
        data-testid="horizontalLayout-img"
        src="https://www.rtlboulevard.nl/sites/default/files/content/images/2018/06/19/ANP-56919532.jpg"
        style={
          Object {
            "width": "30%",
          }
        }
      />
      <div
        className="right-content-div"
        data-testid="horizontalLayout-rightContentDiv"
      >
        <div
          className="label card-title h5"
          data-testid="horizontalLayout-label-title"
        >
          De uitschieters van 2018
        </div>
        <div
          className="label card-title h5"
          data-testid="horizontalLayout-label-date"
        >
          19 juni 2018
        </div>
        <span
          className="layout-bold-title"
          data-testid="horizontalLayout-layoutBoldTitle"
        >
          Trijntje
        </span>
        <span
          className="lead"
          data-testid="horizontalLayout-lead"
        >
          Humberto Tan
        </span>
      </div>
    </div>
  </a>
</div>
`;
